### ISSUE TITLE
Images Become Repeated When Switching to Horizontal or Vertical Grid During Image Stitching

### ISSUE THREAD

**Issue by OkyDooky** [T8RIN/ImageToolbox#1345]
**Describe the bug**
When stitching images, if I just leave it at Horizontal or Vertical, it will behave as expected. But, when I switch to Horizontal Grid or Vertical Grid, it will start displaying duplicates of some images in place of others (see screenshots).
Upon reviewing my screenshots, it appears to just be the last half of the image set that gets repeated for some reason.

**To Reproduce**
Add 8 images to Image Stitching and then switch to Horizontal (or Vertical) Grid mode.

**Expected behavior**
It should take the images as is and just arrange them in a different pattern.

**Screenshots**
![Screenshot_20240917-233203_Image Toolbox](https://github.com/user-attachments/assets/8841dfb8-208c-44fb-8aca-35eb1a14d66f)
![Screenshot_20240917-233335_Image Toolbox](https://github.com/user-attachments/assets/8d70ee3c-8f17-418e-8066-e4158221f455)
![Screenshot_20240917-233526_Image Toolbox](https://github.com/user-attachments/assets/b054bf02-10de-4b76-bbaf-b3343b640007)
For some reason the "Scale small images to large" option broke for me when switching back and forth between layout modes to take these screenshots. If it happens consistently in the future, I'll open an issue for it. Otherwise, it's just to note why they look the way they do in the Grid layouts.

**Smartphone (please complete the following information):**
 - Device: Moto G7 Supra (also happens on my Power variant)
 - OS: 9 (Lineage 20/Android 13 on the Power one)

**Comment by T8RIN** [comment]
App version?

**Comment by OkyDooky** [comment]
52 FOSS.

**Comment by T8RIN** [comment]
What 52 means?

**Comment by T8RIN** [comment]
3.0.0 version?

**Comment by T8RIN** [comment]
Can't reproduce it on my latest build, just no way

**Comment by OkyDooky** [comment]
> What 52 means?

This.
![Screenshot_20240918-140618_Image Toolbox](https://github.com/user-attachments/assets/6036047d-0e40-44ca-9511-2aca50740d62)


> 3.0.0 version?

Yes.

> Can't reproduce on latest build

You mean the latest alpha release? (3.1.0-alpha01)
I will try with it and report back.

**Comment by T8RIN** [comment]
52 is features count, not a version 👀

**Comment by OkyDooky** [comment]
> 52 is features count, not a version 👀

Huh. That's confusing. Thanks for clarifying that. I thought it was an internal versioning system (like what build it was). I'll refer to the x.x.x-whatever version system from now on.

Anyways, I just tried it again with the 3.1.0-alpha01 build and...same problem:

![Screenshot_20240918-141601_Image Toolbox](https://github.com/user-attachments/assets/7e822d4c-9969-4658-9418-92d603db8f52)

I tried looking at the new Collage Maker feature, but it still crops off part of the manga pages, so I don't think it would be able to substitute for this particular situation, sadly. It otherwise loads in all and displays images correctly (no repeats).
![Screenshot_20240918-142323_Image Toolbox](https://github.com/user-attachments/assets/056e3f72-40d1-47c2-a728-5936590f3d00)
![Screenshot_20240918-142327_Image Toolbox](https://github.com/user-attachments/assets/e71f4459-60f7-4025-a2e9-57c7c45629cc)

**Comment by T8RIN** [comment]
You can move and zoom images there

**Comment by T8RIN** [comment]
i will try to repro it again

**Comment by OkyDooky** [comment]
> You can move and zoom images there

Yes, I tried that. The issue is that the aspect ratio of the collage frames is different from the manga pages.
But, in general, I like the features, so far. It was really easy to figure out how to move the images around.

**Comment by T8RIN** [comment]
Still can't reproduce it, but i noticed that after images rearrangement the preview doesn't get updated

**Comment by OkyDooky** [comment]
That's weird. I tried it on a different device (a Galaxy Note 3 T-Mobile variant with Lineage 18/Android 11 installed) using the latest alpha release and I had the same issue. Input 8 images, switch to Horizontal Grid, last 4 images repeated in two rows.
The thing is, on all my devices, the preview result is what gets exported for me. So, the issue winds up in the final image.

I found a workaround by stitching the first 4 images into a horizontal row and exporting, then stitching the last 4 images into a horizontal row and export that, and finally importing both stitches and doing a vertical row. It's more steps, but it gets the same result, for now.

I think I'll try installing an earlier version and see if it starts working. If nothing else, that might provide a clue.
It's still bizarre that I'm having the same issue on three separate devices, each with different Android versions, and it's not reproduceable on your end. Usually these kinds of things are mystery issues that only affects one specific device.

**Comment by OkyDooky** [comment]
I found it.
When I uninstalled the current version and tried an earlier version, it worked like expected. Then I upgraded it to the latest version and...it worked like expected. But then, I toggled off the Add Timestamp setting under the Filename category and...it broke the image stitching. Then I tried it with the Add Timestamp and Formatted Timestamp toggles switched back on and it stitched correctly, no matter what images I threw at it.
Here is a config backup from one of my other devices where I have the settings the way I set them before doing this testing (packed in a ZIP so GitHub would let me upload):
[ImageToolbox Settings Backup.zip](https://github.com/user-attachments/files/17218255/ImageToolbox.Settings.Backup.zip)
